sched/init/nx_start.c

```diff
@@ -628,15 +628,6 @@ void nx_start(void)
       irq_initialize();
     }
 
-  /* Initialize the watchdog facility (if included in the link) */
-
-#ifdef CONFIG_HAVE_WEAKFUNCTIONS
-  if (wd_initialize != NULL)
-#endif
```

Comment:
`wd_initialize` is defined as a weak, so the user is able to override it with a custom initialization procedure.
So, to a certain extent, this is a breaking change, isn't it?

Reply:
> `wd_initialize` is defined as a weak, so the user is able to override it with a custom initialization procedure. So, to a certain extent, this is a breaking change, isn't it?
> 

As I understand, the weak used here want to avoid the linker bring the unused code/data if nobody call wd_xxx function, that's why the code check wd_initialize before invocation.

> It will impact anyone that may be relying on this.

The better approach is modify wd_initialize directly.